apiserver/apiserver.go

```diff
@@ -195,13 +196,19 @@ func newServer(s *state.State, lis net.Listener, cfg ServerConfig) (_ *Server, e
 	if err := srv.updateCertificate(cfg.Cert, cfg.Key); err != nil {
 		return nil, errors.Annotatef(err, "cannot set initial certificate")
 	}
-	logPath := filepath.Join(srv.logDir, "logsink.log")
-	logSinkWriter, err := newLogSinkWriter(logPath)
+
+	logSinkWriter, err := newLogSinkWriter(filepath.Join(srv.logDir, "logsink.log"))
 	if err != nil {
 		return nil, errors.Annotate(err, "creating logsink writer")
 	}
 	srv.logSinkWriter = logSinkWriter
 
+	migratedLogWriter, err := newLogSinkWriter(filepath.Join(srv.logDir, "migrated.log"))
```

Comment:
There is still a problem here in that migrated.log and any rotated copies will be left behind unmanaged. Given that migrations will be fairly infrequent this potentially eats up another 900MB of disk on each controller machine with no real purpose.

I wonder if we even need a file backup of the migrated logs? It really is just a last resort thing if mongodb dies and I don't think it's super important for historical logs. Post migration, any new activity on the new controller will be captured in logsink.log. My gut says not to have a log file for migrated logs (i.e. allow the Writer passed to newLogSinkHandler to be nil)